### PR TITLE
[pickers] Remove  `FieldValueType` in favor of `PickerValueType`

### PR DESCRIPTION
**Pull request by arthurbalduini**
Extracted from #14397 

### Breaking change

- The `FieldValueType` type was renamed `PickerValueType` — [Learn more](https://next.mui.com/x/migration/migration-pickers-v7/#renamed-variables).

### CODE REVIEW

**Review by flaviendelangle — COMMENTED** [review]
The change seems good, thanks for taking care of it!
You should be able to create a codemod in this PR (make sure to use `renameImports` utility) and write the migrate guide (which should be a very basic `diff`)

File: docs/data/migration/migration-pickers-v7/migration-pickers-v7.md [hunk 1]
```diff
@@ -474,3 +474,12 @@ If you have customized those translation keys, you have to update them following
 +  value == null ? null : value.format('hh:mm:ss')
 +);
 ```
+
+## `FieldValueType` renaming
```
Comment (flaviendelangle) [inline docs/data/migration/migration-pickers-v7/migration-pickers-v7.md:478]: We can probably move it to [this section](https://github.com/mui/mui-x/blob/master/docs/data/migration/migration-pickers-v7/migration-pickers-v7.md#renamed-variables).

File: packages/x-codemod/src/util/renameTypes.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  if (parent.type === 'Identifier') {
+    return parent.name;
+  }
+  if (parent.type === 'TSQualifiedName') {
+    return `${getTypeName(parent.left)}.${parent.right.name}`;
+  }
+  return '';
+};
+
+export default function renameTypes(parameters: RenameImportsParameters) {
```
Comment (flaviendelangle) [inline packages/x-codemod/src/util/renameTypes.ts:24]: I'm curious to know why `renameImports` did not work here

**Review by flaviendelangle — APPROVED** [review]
Everything looks good :pray:

**Review by LukasTy — APPROVED** [review]
Good job, LGTM. 👍 

P.S. I have restarted Netlify deployment without cache to hopefully fix it, but the CI status will not be updated until a new CI run is triggered. 😉

**Pull request merged**